### ISSUE TITLE
Not logged in on home

### ISSUE THREAD

**Issue by simon12q** [woocommerce/woocommerce#25138]
When i set 'Shop' page as 'Home' in settings i have problem with login. If i log in on all pages header is changed (my account link etc) but when i go home there look like im not login. Even on shop page which is the same as home im logged in. On the Home page only help Reload(F5), when go to home by logo or link it looks like im not looged in.

**Isolating the problem (mark completed items with an [x]):**
- [x ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```

WordPress environment
--
`
### WordPress Environment ###

WordPress address (URL): 
Site address (URL): 
WC Version: 3.8.1
REST API Version: ✔ 1.0.2
Log Directory Writable: ✔
WP Version: 5.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: pl_PL
External object cache: –

### Server Environment ###

Server Info: Apache/2.2.16 (Debian)
PHP Version: 7.2.11
PHP Post Max Size: 100 MB
PHP Time Limit: 600
PHP Max Input Vars: 1000
cURL Version: 
SUHOSIN Installed: –
MySQL Version: ❌ 5.5.33-0+wheezy1 - We recommend a minimum MySQL version of 5.6. See: WordPress requirements
Max Upload Size: 100 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.8.1
WC Database Prefix: vh2_
MaxMind GeoIP Database: ❌ The MaxMind GeoIP Database does not exist - Geolocation will not function. You can download and install it manually from https://dev.maxmind.com/geoip/geoip2/geolite2/ to the path: . Scroll down to "Downloads" and download the "MaxMind DB binary
gzipped" file next to "GeoLite2 Country". Please remember to uncompress GeoLite2-Country_xxxxxxxx.tar.gz and upload the GeoLite2-Country.mmdb file only.

Total Database Size: 9.90MB
Database Data Size: 8.34MB
Database Index Size: 1.56MB
vh2_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
vh2_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
vh2_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_comments: Data: 0.06MB + Index: 0.09MB + Engine InnoDB
vh2_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_options: Data: 3.23MB + Index: 0.06MB + Engine InnoDB
vh2_postmeta: Data: 1.48MB + Index: 0.27MB + Engine InnoDB
vh2_posts: Data: 1.50MB + Index: 0.13MB + Engine InnoDB
vh2_revslider_css: Data: 0.13MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_css_bkp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_layer_animations: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_layer_animations_bkp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_navigations: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_navigations_bkp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_sliders: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_sliders_bkp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_slides: Data: 1.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_slides_bkp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_static_slides: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_revslider_static_slides_bkp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
vh2_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
vh2_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
vh2_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
vh2_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
vh2_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
vh2_woodmart_wishlists: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
vh2_woodmart_wishlist_products: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (10) ###

Contact Form 7: by Takayuki Miyoshi – 5.1.5
WPBakery Page Builder: by Michael M - WPBakery.com – 6.0.5
MC4WP: Mailchimp for WordPress: by ibericode – 4.7.2
Slider Revolution: by ThemePunch – 6.1.5
Safe SVG: by Daryll Doyle – 1.9.6
User Role Editor: by Vladimir Garagulya – 4.52.1
WooCommerce Admin: by WooCommerce – 0.22.0
Role Based Price For WooCommerce: by Varun Sridharan – 3.3.6 – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.8.1
Woodmart Core: by  – 1.0.20

### Inactive Plugins (0) ###


### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #10 - /
Cart: #11 - /cart/
Checkout: #12 - /checkout/
My account: #13 - /my-account/
Terms and conditions: ❌ Page visibility should be public

### Theme ###

Name: Woodmart
Version: 4.2.2
Author URL: http://themeforest.net/user/xtemos
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: woodmart/woocommerce/archive-product.php
woodmart/woocommerce/cart/cart-item-data.php
woodmart/woocommerce/cart/cart.php
woodmart/woocommerce/cart/cross-sells.php
woodmart/woocommerce/cart/mini-cart.php
woodmart/woocommerce/checkout/form-checkout.php
woodmart/woocommerce/content-product.php
woodmart/woocommerce/content-product_cat.php
woodmart/woocommerce/content-single-product.php
woodmart/woocommerce/content-widget-product.php
woodmart/woocommerce/global/breadcrumb.php
woodmart/woocommerce/global/form-login.php
woodmart/woocommerce/global/quantity-input.php
woodmart/woocommerce/loop/add-to-cart.php
woodmart/woocommerce/loop/loop-end.php
woodmart/woocommerce/loop/loop-start.php
woodmart/woocommerce/loop/no-products-found.php
woodmart/woocommerce/loop/orderby.php
woodmart/woocommerce/loop/pagination.php
woodmart/woocommerce/loop/sale-flash.php
woodmart/woocommerce/myaccount/form-login.php
woodmart/woocommerce/product-searchform.php
woodmart/woocommerce/single-product/add-to-cart/grouped.php
woodmart/woocommerce/single-product/add-to-cart/variable.php
woodmart/woocommerce/single-product/meta.php
woodmart/woocommerce/single-product/product-image.php
woodmart/woocommerce/single-product/product-thumbnails.php
woodmart/woocommerce/single-product/rating.php
woodmart/woocommerce/single-product/related.php
woodmart/woocommerce/single-product/sale-flash.php
woodmart/woocommerce/single-product/tabs/tabs.php
woodmart/woocommerce/single-product/title.php
woodmart/woocommerce/single-product/up-sells.php
woodmart/woocommerce/single-product.php


### Action Scheduler ###

Complete: 73
Oldest: 2019-11-14 09:37:12 +0100
Newest: 2019-11-29 09:45:25 +0100

Pending: 1
Oldest: 2019-11-29 10:40:32 +0100
Newest: 2019-11-29 10:40:32 +0100

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –

`


```
</details>

**Comment by juliaamosova** [comment]
@simon12q thanks for opening an issue. However, it is not clear how to reproduce it. 

Please provide the details of your settings (screenshots will work) and the exact steps to reproduce the issue. 

>If i log in on all pages header is changed (my account link etc)

How does it change?

**Comment by simon12q** [comment]
It;s simple to reproduce.
Clean install of newest wordpress + newest woocommerce without any additional plugins.
Set up default 'shop' page. Then log in and go to /shop page. It's looks like cached. Need to F5 or Ctrl+F5 to see admin bar.

**Comment by juliaamosova** [comment]
>Clean install of newest wordpress + newest woocommerce without any additional plugins.
Set up default 'shop' page. Then log in and go to /shop page. It's looks like cached. Need to F5 or Ctrl+F5 to see admin bar.

I tested further according to your instructions but I don't see any issue. Here is a screencast I made on a brand new install:

![](http://g.recordit.co/jh0BE7aFnx.gif)

>It's looks like cached. 

I don't understand what it means. A screenshot would be helpful.
